include/physics/diff_physics.h

```diff
@@ -303,6 +303,53 @@ class DifferentiablePhysics
   virtual bool nonlocal_mass_residual (bool request_jacobian,
                                        DiffContext &c);
 
+  /**
+   * Subtracts a damping vector contribution on \p elem from
+   * elem_residual.
+   *
```

Comment:
Could we document this better?  Likewise for the definition of mass_residual in second order problems?